scripts/build_windows.ps1

```diff
@@ -0,0 +1,130 @@
+        & "${script:SignTool}" sign /v /fd sha256 /t http://timestamp.digicert.com /f "${env:OLLAMA_CERT}" `
+            /csp "Google Cloud KMS Provider" /kc ${env:KEY_CONTAINER} ollama.exe
+        if ($LASTEXITCODE -ne 0) { exit($LASTEXITCODE)}
+    }
+}
+
+function buildApp() {
+    write-host "Building Ollama App"
+    cd "${script:SRC_DIR}\app"
+    & go build "-ldflags=-H windowsgui -w -s ""-X=github.com/jmorganca/ollama/version.Version=$script:VERSION"" ""-X=github.com/jmorganca/ollama/server.mode=release""" .
+    if ($LASTEXITCODE -ne 0) { exit($LASTEXITCODE)}
+    if ("${env:KEY_CONTAINER}") {
+        & "${script:SignTool}" sign /v /fd sha256 /t http://timestamp.digicert.com /f "${env:OLLAMA_CERT}" `
+            /csp "Google Cloud KMS Provider" /kc ${env:KEY_CONTAINER} app.exe
+        if ($LASTEXITCODE -ne 0) { exit($LASTEXITCODE)}
+    }
+}
+
+function gatherDependencies() {
+    write-host "Gathering runtime dependencies"
+    cd "${script:SRC_DIR}"
+    rm -ea 0 -recurse -force -path "${script:DEPS_DIR}"
+    md "${script:DEPS_DIR}" -ea 0 > $null
+
+    # TODO - this varies based on host build system and MSVC version - drive from dumpbin output
+    # currently works for Win11 + MSVC 2019 + Cuda V11
+    cp "${env:VCToolsRedistDir}\x64\Microsoft.VC*.CRT\msvcp140.dll" "${script:DEPS_DIR}\"
+    cp "${env:VCToolsRedistDir}\x64\Microsoft.VC*.CRT\vcruntime140.dll" "${script:DEPS_DIR}\"
+    cp "${env:VCToolsRedistDir}\x64\Microsoft.VC*.CRT\vcruntime140_1.dll" "${script:DEPS_DIR}\"
+
+    cp "${script:NVIDIA_DIR}\cudart64_*.dll" "${script:DEPS_DIR}\"
+    cp "${script:NVIDIA_DIR}\cublas64_*.dll" "${script:DEPS_DIR}\"
+    cp "${script:NVIDIA_DIR}\cublasLt64_*.dll" "${script:DEPS_DIR}\"
+
+    cp "${script:SRC_DIR}\app\ollama_welcome.ps1" "${script:SRC_DIR}\dist\"
+    if ("${env:KEY_CONTAINER}") {
+        write-host "about to sign"
+        foreach ($file in (get-childitem "${script:DEPS_DIR}/cu*.dll") + @("${script:SRC_DIR}\dist\ollama_welcome.ps1")){
+            write-host "signing $file"
+            & "${script:SignTool}" sign /v /fd sha256 /t http://timestamp.digicert.com /f "${env:OLLAMA_CERT}" `
+                /csp "Google Cloud KMS Provider" /kc ${env:KEY_CONTAINER} $file
+            if ($LASTEXITCODE -ne 0) { exit($LASTEXITCODE)}
+        }
+    }
+
+}
+
+function buildInstaller() {
+    write-host "Building Ollama Installer"
+    cd "${script:SRC_DIR}\app"
+    $env:PKG_VERSION=$script:PKG_VERSION
+    if ("${env:KEY_CONTAINER}") {
+        & "${script:INNO_SETUP_DIR}\ISCC.exe" /SMySignTool="${script:SignTool} sign /fd sha256 /t http://timestamp.digicert.com /f ${env:OLLAMA_CERT} /csp `$qGoogle Cloud KMS Provider`$q /kc ${env:KEY_CONTAINER} `$f" .\ollama.iss
+    } else {
+        & "${script:INNO_SETUP_DIR}\ISCC.exe" .\ollama.iss
+    }
+    if ($LASTEXITCODE -ne 0) { exit($LASTEXITCODE)}
+}
+
+try {
+    checkEnv
+    buildOllama
+    buildApp
+    gatherDependencies
+    buildInstaller
```

Comment:
does this also export the ollama windows binary? i.e. the non-app windows executable? macos ships both the app `Ollama.app` and the non-app binary `ollama-darwin`. it would be nice to have this option for windows as well

Reply:
This is addressed in #2516